src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -40,23 +41,35 @@ public void init() {
   }
 
   /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
+   * Responds with a JSON representation of {@link Message} data for logged-in user and another user. Responds with
+   * an empty array if the user is not logged in or another user is not provided.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
 
-    String user = request.getParameter("user");
-
-    if (user == null || user.equals("")) {
-      // Request is invalid, return empty array
-      response.getWriter().println("[]");
+    // Get logged-in user and other user
+    Optional<String> loggedInUser = Optional.ofNullable(null);
+    Optional<String> otherUser = Optional.ofNullable(request.getParameter("user"));
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      loggedInUser = Optional.ofNullable(userService.getCurrentUser().getEmail());
+    }
+    
+    // Request is invalid, return empty array
+    if (loggedInUser.orElse("").isEmpty()) {
+      response.setStatus(401);
+      response.getWriter().println("Error: Unauthorized access");
```

Comment:
I would strongly recommend making the error response a JSON object. That will allow you to add more properties/information to the error in the future, but that change can be made in a separate pull request.